packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -439,6 +440,22 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         40 + 30 * baselineProps.rows.length,
       );
     });
+
+    // A function test counterpart of ScrollbarOverflowVerticalSnap.
+    it('should not have a horizontal scrollbar if not needed', () => {
+      const TestCase = () => {
+        const data = useData(100, 1);
```

Comment:
this is a simple case. The test should cover when we have the total column width that is within the container width - scroll width ie 
container width of 500, total column width of 495 no scroll bar if no vertical scrollbar, also we could check with a total width of 485 so no scroll bar even if there is a vertical one
Does that make sense?

Reply:
It does make sense, for another PR. For now, it fixes this condition which is a step forward.